pyrmq/tests/test_consumer.py

```diff
@@ -116,3 +116,43 @@ def error_callback(*args):
 
             wait_for_result(response, [1])
             consumer.close()
+
+
+def should_get_message_with_higher_priority(priority_session: Publisher):
+    expected = {"test": "test", "priority": 0}
+    for i in range(0, 10):
+        body = {"test": "test", "priority": i}
+        priority_session.publish(body, priority=i)
+    response = {}
+
+    def callback(data):
+        if not response:
+            """
+            Checks first message if it has the highest priority. Current max priority is 5
+            Messages with priority 5 and above will be treated as priority 5.
+            Thus, they will be queued after the first message received with the highest priority
+            Example 1:
+                message1 = {"test": "test", "priority": 10}
+                message2 = {"test": "test", "priority": 5}
+                message3 = {"test": "test", "priority": 8}
+            message1 will be on the queue first, followed by message2 then message3
+            Example 2:
+                message1 = {"test": "test", "priority": 4}
+                message2 = {"test": "test", "priority": 8}
+                message3 = {"test": "test", "priority": 5}
+            message2 will be on the queue first, followed by message3 then message1
+            """
+            first_expected = {"test": "test", "priority": 5}
```

Comment:
Shouldn't this be 10 because it has the highest priority?

Reply:
The max priority set, through `x-max-priority` is 5. Meaning every priority number after 5 already counts as max priority. So 5 is the first consumed.

Reply:
How are we sure that it does not treat priority number before 5 as 5?

Maybe better if we have different data/message depending on the priority.

Reply:
I added a check at the bottom that the last message to be consumed is the lowest priority, in this case priority 0. FIFO still stands in the queue but with max priority it will treat messages with priority 5 and above as priority 5. If queue receives a message with priority 10 first before any priority >=5, it will be the first one to be consumed.

Reply:
Oh okay. So it changes all priority > 5 to 5 first, then, it will consume the priority 5's based on FIFO.